index.js

```diff
@@ -27,93 +23,73 @@ module.exports = function (homebridge) {
-      id17: { description: "Sidebrushes malfunctioned. Reboot the system." },
-      id18: { description: "Fan malfunctioned. Reboot the system." },
-      id19: { description: "The docking station is not connected to power." },
-      id20: { description: "unkown" },
-      id21: {
-        description:
-          "Please make sure that the top cover of the laser distance sensor is not pinned.",
-      },
-      id22: { description: "Please wipe the dock sensor." },
-      id23: {
-        description: "Make sure the signal emission area of dock is clean.",
-      },
+      id1: { description: 'Try turning the orange laserhead to make sure it isnt blocked.' },
+      id2: { description: 'Clean and tap the bumpers lightly.' },
+      id3: { description: 'Try moving the vacuum to a different place.' },
+      id4: { description: 'Wipe the cliff sensor clean and move the vacuum to a different place.' },
+      id5: { description: 'Remove and clean the main brush.' },
+      id6: { description: 'Remove and clean the sidebrushes.' },
+      id7: { description: 'Make sure the wheels arent blocked. Move the vacuum to a different place and try again.' },
+      id8: { description: 'Make sure there are no obstacles around the vacuum.' },
+      id9: { description: 'Install the dustbin and the filter.' },
+      id10: { description: 'Make sure the filter has been tried or clean the filter.' },
+      id11: { description: 'Strong magnetic field detected. Move the device away from the virtual wall and try again' },
+      id12: { description: 'Battery is low, charge your vacuum.' },
+      id13: { description: 'Couldnt charge properly. Make sure the charging surfaces are clean.' },
+      id14: { description: 'Battery malfunctioned.' },
+      id15: { description: 'Wipe the wall sensor clean.' },
+      id16: { description: 'Use the vacuum on a flat horizontal surface.' },
+      id17: { description: 'Sidebrushes malfunctioned. Reboot the system.' },
+      id18: { description: 'Fan malfunctioned. Reboot the system.' },
+      id19: { description: 'The docking station is not connected to power.' },
+      id20: { description: 'unkown' },
+      id21: { description: 'Please make sure that the top cover of the laser distance sensor is not pinned.' },
+      id22: { description: 'Please wipe the dock sensor.' },
+      id23: { description: 'Make sure the signal emission area of dock is clean.' }
     };
   }
 
   constructor(log, config) {
     this.log = log;
     this.config = config;
-    this.config.name = config.name || "Roborock vacuum cleaner";
-    this.config.cleanword = config.cleanword || "cleaning";
+    this.config.name = config.name || 'Roborock vacuum cleaner';
+    this.config.cleanword = config.cleanword || 'cleaning';
     this.config.delay = config.delay || false;
     this.services = {};
 
     // Used to store the latest state to reduce logging
     this.cachedState = new Map();
 
+    //Changed by Mai
+    this.roomsToClean = [];
+
```

Comment:
Actual change to include in a PR

Reply:
The idea is to use an array to store room ids to be cleaned, and the switches for each room is basically add/remove id from this list. When start clean, it checks the list. If the list contain rooms, it does room cleaning, otherwise full cleaning. Similarly resume also call the right operation based on if the list is empty or not. When cleaning is finished, all ids are removed from the list.